app/controllers/notification_settings_controller.rb

```diff
@@ -0,0 +1,19 @@
+class NotificationSettingsController < ApplicationController
```

Comment:
Don't do this in a separate controller, but use the Profile + Question controllers instead.

You can create the form for the profile for example like this:

```ruby
# profiles_controller.rb

def edit
  @profile.notification_setting ||= @profile.notification_setting.build
  # ...
end
```

Then in the form:

```erb
<!-- app/views/profiles/_form.html.erb -->
<%= form_for(@profile, ... %>
  <!-- first part of the form -->

  <%= fields_for(:notification_setting) do |n| %>
    <%= n.check_box :send_emails %> 
    <!-- etc.. -->
  <% end %>
<% end %>
```

Then, in your profiles controller, allow these things to be posted:

```ruby
  def profile_params
    params.require(:profile).permit(:id, ..., :notificaton_setting_attributes: { :send_emails })
  end
```

And finally, tell the Profile model to accept nested params for the notification settings:

```ruby
class Profile
  accepts_nested_attributes_for :notification_setting
  # ...
end
```